server/meca/routes.test.js

```diff
@@ -107,4 +107,24 @@ describe('MECA HTTP callback handler', () => {
       .expect(500, { error: 'Manuscript not found' })
     expect(logger.error).toHaveBeenCalled()
   })
+
+  it('triggers the auditLog.save', async () => {
+    const manuscript = await new Manuscript({ createdBy: userId }).save()
+    const request = makeApp()
+    expect(manuscript.status).toBe(Manuscript.statuses.INITIAL)
+    await request
+      .post(`/meca-result/${manuscript.id}`)
+      .set('Authorization', `Bearer ${apiKey}`)
+      .send({ result: 'success' })
+      .expect(204)
+    const updatedManuscript = await Manuscript.find(manuscript.id, userId)
+    expect(updatedManuscript.status).toBe(
+      Manuscript.statuses.MECA_IMPORT_SUCCEEDED,
+    )
+    const audits = await AuditLog.all()
+    expect(audits).toHaveLength(2)
+    expect(audits[0].value).toBe('MECA_IMPORT_SUCCEEDED')
+    expect(audits[1].objectType).toBe('ejp.id')
+    expect(audits[1].objectId).toBe(manuscript.id)
```

Comment:
I don't think we should have a separate log entry for this alongside the existing one.

Currently, the AuditLog is meant to show changes made to the database. Recording the `ejp.id` in a separate AuditLog entry goes against that pattern.

Instead, I think we should be able to contain the `ejp.id` within the initial AuditLog entry. We can put it within either the `value` field, or add an additional column for `userType` to store a value like `EJP`, and the `userId` field would contain the EJP id.

@diversemix would be good to get your opinion on this.

Reply:
Current AuditLog entries are as follows:
```
[ AuditLog {
        id: 'a9d27bc9-40d6-4736-851b-3bb35bc67b8a',
        created: 2019-03-04T14:47:21.311Z,
        userId: null,
        objectId: '5db164cc-9c52-4948-8f0f-323f53e6bd49',
        objectType: 'manuscript.status',
        updated: 2019-03-04T14:47:21.311Z,
        value: 'MECA_IMPORT_SUCCEEDED',
        action: 'UPDATED' },
      AuditLog {
        id: 'fb5ac23b-7f1c-4345-b2c9-f99368b035d5',
        created: 2019-03-04T14:47:21.384Z,
        userId: null,
        objectId: '5db164cc-9c52-4948-8f0f-323f53e6bd49',
        objectType: 'ejp.id',
        updated: 2019-03-04T14:47:21.384Z,
        value: null,
        action: 'UPDATED' } ]
```

Reply:
agree, let's hold this on till have a chat with Peter, to see what he meant

Reply:
Ok, so my point of view - happy for it to be challenged....

First AuditLog entry - this should be done in the REST route of the POST `/meca-result` to record the fact an external entity has called into the application.

Second AuditLog entry I would expect to be the recording of that consequences of that request - in this case on the Manuscript's `state`

Reply:
The first AuditLog entry is captured by our logging, and I think that's the correct place for us to be logging that information. 

I prefer to have the AuditLog used to track database changes, and the database to contain all the information/IDs we need to then use in other systems to continue tracing any external services we're using, like S3.

Reply:
Talking with @hdrury1 ... It is a product requirement to be able to track (audit) the lifecycle of a manuscript for the lifetime of the product (i.e. beyond the 30 day limit of the logs). Ideally we do want to also track S3 and SFTP interactions as well in the future.